### PR TITLE
Add logging settings

### PR DESCRIPTION
**Pull request by Robert-Steiner**
Summary:
- adds the possibility to define `logging`in the config file

### CODE REVIEW

File: rust/src/settings.rs [hunk 1]
```diff
@@ -15,14 +17,32 @@ pub enum SettingsError {
...
     pub api: ApiSettings,
     #[validate]
     pub pet: PetSettings,
     pub mask: MaskSettings,
+    pub log: LoggingSettings,
+}
+
+impl Settings {
+    /// Loads and validates the coordinator settings via a configuration file.
+    /// Fails when the loading of the configuration file or its validation failed.
```
Comment (janpetschexain) [inline rust/src/settings.rs:33]:
```suggestion
    /// Loads and validates the coordinator settings via a configuration file.
    ///
    /// # Errors
    /// Fails when the loading of the configuration file or its validation failed.
```

File: configs/config.toml [hunk 2]
```diff
@@ -1,3 +1,6 @@
+[log]
+filter = "xain_fl=debug,info"
```
Comment (janpetschexain) [inline configs/config.toml:2]: i thought that higher priority levels would automatically include lower priority levels when logging, like `error > warn > info > debug > trace` for the [level filters](https://docs.rs/tracing-subscriber/0.2.6/tracing_subscriber/filter/struct.LevelFilter.html). is that not the case?

Comment (Robert-Steiner) [reply configs/config.toml:2]: That is right. However, a filter allows us to define multiple logging directives. 
So `xain_fl=debug` means: turn on `debug` logging for the `xain_fl` module and 
`info` means turn on `info` logging for all modules.

Comment (janpetschexain) [reply configs/config.toml:2]: ahh true, i missed that only the first level after the filter actually applies to the filter.

**Pull request merged**